### PR TITLE
fix(dig): dig not working

### PR DESCRIPTION
**Pull request by kishore03109**
## Problem

We do have some checks in the backend to check prior to any site launch that there are AAAA records/CAA records present. In this case, the dig commands to check the AAAA records [failed](https://ogp.datadoghq.com/logs?query=service%3Aisomer%20env%3Aproduction%20%22An%20error%20occurred%20while%20performing%20dig%20for%20domain%3A%22%20&cols=host%2Cservice&fromUser=true&index=%2A&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&from_ts=1711457521321&to_ts=1711543921321&live=true).

The reason for above is that we were using a library called `node-dig-dns`. This called the dig command [directly](https://github.com/StephanGeorg/node-dig-dns/blob/master/src/index.js#L78) at a system level. However, our docker container does not have the dig command out of the box. This resulted in the existence of AAAA records not being caught.

We are also codifying a check for CAA records and ensuring that if there exist at least one  caa record and it uses our redirection service, it should have letsencrypt as one of the caa record. 

To prevent accidental commits to live indirection repo during dev, also adding a check to only commit to the indirection repository iff it is in prod environment. 
 
## Solution

just use node's dns resolver directly. this way we dont have to depend on an external library's implementation of node and dont have to install unnecessary deps in the docker.

remove dep introduced in #1244 

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [x] No - this PR is backwards compatible with ALL of the following feature flags in this [doc](https://www.notion.so/opengov/Existing-feature-flags-518ad2cdc325420893a105e88c432be5)

**Features**:
Screenshot 2024-03-27 at 11 21 44 PM

### Manual test (not to be copied over to deployment notes)

- [ ] add these lines of code at the end of server.js 
```
const formResponses = [
  {
    submissionId: "",
    requesterEmail: "kishore@open.gov.sg",
    repoName: "kishore-test-dev-emil",
    primaryDomain: "google.com",
    redirectionDomain: "www.google.com",
    agencyEmail: "kishore@open.gov.sg",
  },
]

formsgSiteLaunchRouter.handleSiteLaunchResults(formResponses, "test")
```  

- [ ] Assert that the email comes out to

### CODE REVIEW

**Comment by kishore03109** [comment]
* **#1246** <a href="https://app.graphite.dev/github/pr/isomerpages/isomercms-backend/1246?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kishore03109 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/routes/formsg/formsgSiteLaunch.ts [hunk 1]
```diff
@@ -225,23 +227,91 @@ export class FormsgSiteLaunchRouter {
...
+        domain: launchResult.primaryDomainSource,
+        type: "AAAA",
+        value: record,
+      }))
+    } catch (e: any) {
+      if (e.code && e.code === "ENODATA") {
+        logger.info(
+          `Unable to get dig response for domain: ${launchResult.primaryDomainSource}. Skipping check for AAAA records`
+        )
+      }
```
Comment (seaerchin) [inline src/routes/formsg/formsgSiteLaunch.ts:252]: could we change the `any`? separately, why are we not logging the `error` here? should it be `info` or should we set to `warn`?

File: src/routes/formsg/formsgSiteLaunch.ts [hunk 2]
```diff
@@ -225,23 +227,91 @@ export class FormsgSiteLaunchRouter {
...
+    } catch (e: any) {
+      if (e.code && e.code === "ENODATA") {
+        logger.info(
+          `Unable to get dig response for domain: ${launchResult.primaryDomainSource}. Skipping check for CAA records`
         )
-        return null
-      })
+      }
+      throw e
+    }
```
Comment (seaerchin) [inline src/routes/formsg/formsgSiteLaunch.ts:313]: ssee above comment

File: src/types/siteLaunch.ts [hunk 3]
```diff
@@ -40,6 +40,19 @@ export interface SiteLaunchMessage {
...
+  SiteLaunchMessage,
+  | "status"
+  | "statusMetadata"
+  | "redirectionDomain"
+  | "appId"
+  | "requestorEmail"
+  | "agencyEmail"
+> & {
+  redirectionDomainSource?: string
+  redirectionDomainTarget?: string
```
Comment (seaerchin) [inline src/types/siteLaunch.ts:53]: is this just to facilitate the initial construction of `successResult` and setting the props later? it seems like we'll always have the properties (we throw on error and we have a default) so maybe we should set this as required to better reflect reality

File: src/routes/formsg/formsgSiteLaunch.ts [hunk 4]
```diff
@@ -225,23 +227,91 @@ export class FormsgSiteLaunchRouter {
...
+    addAWSACMCertCAA: boolean
+    addLetsEncryptCAA: boolean
+  }> => {
+    try {
+      const caaDigResponses: CaaRecord[] = await dnsPromises.resolveCaa(
+        launchResult.primaryDomainSource
+      )
+
+      if (caaDigResponses.length > 0) {
+        const caaRecords = caaDigResponses
```
Comment (seaerchin) [inline src/routes/formsg/formsgSiteLaunch.ts:269]: should we flip it so the default case is up top?

**Review by seaerchin — COMMENTED** [review]
since we're removing the external dependency here, should we also update the `dockerfile` to remove the installation of `bind-tools`?

File: src/types/siteLaunch.ts (hunk 3 above)
Comment (kishore03109) [reply src/types/siteLaunch.ts:53]: no ah no all site launches use redirection server leh

File: src/routes/formsg/formsgSiteLaunch.ts (hunk 4 above)
Comment (kishore03109) [reply src/routes/formsg/formsgSiteLaunch.ts]: [5e7d5e9](https://github.com/isomerpages/isomercms-backend/pull/1246/commits/5e7d5e9ca075851f0042b7bfc43852c8812bcf53)

File: src/routes/formsg/formsgSiteLaunch.ts (hunk 2 above)
Comment (kishore03109) [reply src/routes/formsg/formsgSiteLaunch.ts]: [5e7d5e9](https://github.com/isomerpages/isomercms-backend/pull/1246/commits/5e7d5e9ca075851f0042b7bfc43852c8812bcf53)

File: src/routes/formsg/formsgSiteLaunch.ts (hunk 1 above)
Comment (kishore03109) [reply src/routes/formsg/formsgSiteLaunch.ts]: added type guard [5e7d5e9](https://github.com/isomerpages/isomercms-backend/pull/1246/commits/5e7d5e9ca075851f0042b7bfc43852c8812bcf53)

File: src/types/nodeError.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export function isErrnoException(obj: any): obj is NodeJS.ErrnoException {
+  return (
+    obj &&
+    (typeof obj.errno === "number" || obj.errno === undefined) &&
+    (typeof obj.code === "string" || obj.code === undefined) &&
+    (typeof obj.path === "string" || obj.path === undefined) &&
+    (typeof obj.syscall === "string" || obj.syscall === undefined)
+  )
+}
```
Comment (seaerchin) [inline src/types/nodeError.ts:9]:
```suggestion
export function isErrnoException(obj: unknown): obj is NodeJS.ErrnoException {
  const err = obj as NodeJS.ErrnoException
  return (
    err &&
    (typeof err.errno === "number" || err.errno === undefined) &&
    (typeof err.code === "string" || err.code === undefined) &&
    (typeof err.path === "string" || err.path === undefined) &&
    (typeof err.syscall === "string" || err.syscall === undefined)
  )
}
```

**Review by seaerchin — APPROVED** [review]
lgtm, just accept suggestion to remove `any`

**Pull request merged**